leader-election-impl/src/main/java/com/palantir/paxos/SingleLeaderPinger.java

```diff
@@ -88,20 +89,48 @@ public LeaderPingResult pingLeaderWithUuid(UUID uuid) {
         if (!suspectedLeader.isPresent()) {
             return LeaderPingResults.pingReturnedFalse();
         }
-
         LeaderPingerContext<PingableLeader> leader = suspectedLeader.get();
 
         MultiplexingCompletionService<LeaderPingerContext<PingableLeader>, PingResult> multiplexingCompletionService
                 = MultiplexingCompletionService.createFromCheckedExecutors(leaderPingExecutors);
+        try {
+            return actuallyPingLeaderWithUuid(multiplexingCompletionService,
+                    uuid,
+                    leader,
+                    leader.pinger()::pingV2);
+        } catch (ExecutionException e) {
+            log.warn("Could not ping the leader using pingV2 endpoint, ", e);
+        }
+
+        try {
+            return actuallyPingLeaderWithUuid(multiplexingCompletionService,
+                    uuid,
+                    leader,
+                    () -> getPingResultFromLegacyEndpoint(leader));
+        } catch (ExecutionException ex) {
+            return LeaderPingResults.pingCallFailure(ex.getCause());
+        }
```

Comment:
My main concern with this is that for users who may not have access to ping v2, this is going to make performance of ping worse by a full RPC. It's not really the hottest codepath so this may not necessarily be terrible, but is there something we can do to avoid repeatedly trying if there is no ping v2?